postfix/files/main.cf

```diff
@@ -1,17 +1,7 @@
 {%- from "postfix/map.jinja" import postfix with context -%}
 {%- set config = salt['pillar.get']('postfix:config', {}) -%}
 
-{%- if not salt['pillar.get']('postfix:mapping', False) %}
-{#- Let the user configure mapping manually. -#}
-{%- set processed_parameters = [] %}
-{%- else -%}
-{#- TODO: alias_maps probably belongs here, too: #}
-{%- set processed_parameters = [
-        'virtual_alias_maps',
-        'smtp_sasl_password_maps',
-        'sender_canonical_maps',
-    ] %}
-{%- endif -%}
+{%- set processed_parameters = salt['pillar.get']('postfix:mapping', {}).keys() | list %}
```

Comment:
@alxwr I don't think you need additional `list` filter. The `keys()` method will always return a new list.

Reply:
In Python 3.6 `keys()` returns a `dict_keys` object, so the `list` is mandatory.

Without `| list`:

```
jinja2.exceptions.UndefinedError: 'dict_keys object' has no attribute 'append'
```

As far as I understand it: In Python 3.6 `keys()` of an `OrderedDict` object returns a view (which may change) on the keys of the object.

Reply:
Oh, thanks for pointing me out to Python 3.6 specifics, I wasn't aware of that.

However, still think it looks weird. As I can tell from the code, the `processed_parameters` used only once to verify the key exists. No need for such complexity.
We could avoid making the list at all:
```jinja
{%- set processed_parameters = salt['pillar.get']('postfix:mapping', {}) %}
```
Then update the key instead of appending at line 14:
```jinja
{%- do processed_parameters.update({parameter: value}) %}
```
Finally, at line 130 the `if` statement will work the same way.

Reply:
@vutny Thanks for your suggestions regarding the elegance of the code. I appreciate that.

`processed_parameters` was a list, therefore I kept it that way. In my mind it makes more sense to manage a list of parameter _names_ than to needlessly copy their values. `postfix:mapping`'s contents may be quite large. Therefore I suggest we stick with the list.

I disagree with using a dict instead of a list, but if this is non-negotiable, I'll give way. :-)

Reply:
@alxwr No worries, I would just like to make things here as portable as possible because of two reasons:
* Generally using Python methods directly is undesirable, unless there's no other clear way.
  As we've already learned, different Python version have "smaaaall" differences. And the Jinja code is not really Python, it is a layer on top. So it is better to go with Jinja semantics as much as possible in common situations.
* The most heavy call here is `pillar.get`. It needs to fork minion process, talk to the master via message bus and network layer, receive probably megabytes back, cache it, deserialize and finally return back to the caller.
  Lots of stuff going on under the hood. Comparing to that, storing even a big dict in temporary memory is nothing.

If your concern is that holding the values is unnecessary, we could use [pillar.keys](https://docs.saltstack.com/en/latest/ref/modules/all/salt.modules.pillar.html#salt.modules.pillar.keys) function! Nothing else need to be changed then.

Reply:
@vutny Then we're on the same page. :-) Thanks for the thorough explanation!

Why didn't I think of `pillar.keys` in the first place? O.o
Thanks for the review and the suggestion! I updated the PR.

The `| list` is needed because of Py3.6: "jinja2.exceptions.UndefinedError: 'dict_keys object' has no attribute 'append'".